Deprecated test is failing

## Reason
See https://civet.inl.gov/job/2282288/

```
In file included from /data/civet2/build/salamander/build/unity_src/userobjects_Unity.C:8:
/data/civet2/build/salamander/src/userobjects/ParticlesPerElementInitializer.C: In member function 'virtual std::vector<InitialParticleData> ParticlesPerElementInitializer::getParticleData() const':
/data/civet2/build/salamander/src/userobjects/ParticlesPerElementInitializer.C:85:3: error: 'UniquePtr' was not declared in this scope; did you mean 'UniqueStorage'?
   85 |   UniquePtr<FEBase> fe = FEBase::build(_mesh_dimension, fe_type);
      |   ^~~~~~~~~
      |   UniqueStorage
/data/civet2/build/salamander/src/userobjects/ParticlesPerElementInitializer.C:85:19: error: expected primary-expression before '>' token
   85 |   UniquePtr<FEBase> fe = FEBase::build(_mesh_dimension, fe_type);
      |                   ^
/data/civet2/build/salamander/src/userobjects/ParticlesPerElementInitializer.C:85:21: error: 'fe' was not declared in this scope
   85 |   UniquePtr<FEBase> fe = FEBase::build(_mesh_dimension, fe_type);
      |                     ^~
make: *** [/data/civet2/build/salamander/moose/framework/build.mk:150: /data/civet2/build/salamander/build/unity_src/userobjects_Unity.x86_64-pc-linux-gnu.opt.lo] Error 1
```

## Design
Use `std::unique_ptr` instead of `UniquePtr`, which is deprecated in libmesh.

## Impact
Allows salamander to compile without deprecated methods.

--- a/src/userobjects/ParticlesPerElementInitializer.C
+++ b/src/userobjects/ParticlesPerElementInitializer.C
@@ -82,7 +82,7 @@ ParticlesPerElementInitializer::getParticleData() const
   // setting up this to be able to map from reference elements to the physical elements
   ArbitraryQuadrature arbitrary_qrule = ArbitraryQuadrature(_mesh_dimension, FIRST);
   FEType fe_type = FEType(CONSTANT, MONOMIAL);
-  UniquePtr<FEBase> fe = FEBase::build(_mesh_dimension, fe_type);
+  std::unique_ptr<FEBase> fe = FEBase::build(_mesh_dimension, fe_type);
   fe->attach_quadrature_rule(&arbitrary_qrule);
   fe->get_xyz();
   // random number generator to be reseeded on each element

